FirebaseStorage/Sources/StorageMetadata.swift

```diff
@@ -162,80 +105,199 @@ import FirebaseStorageInternal
   /**
    * A reference to the object in Firebase Storage.
    */
-  @objc public let storageReference: StorageReference?
+  // @objc public let storageReference: StorageReference?
 
   /**
    * Creates a Dictionary from the contents of the metadata.
    * @return A Dictionary that represents the contents of the metadata.
    */
-  @objc open func dictionaryRepresentation() -> [String: Any] {
-    return impl.dictionaryRepresentation()
+  @objc open func dictionaryRepresentation() -> [String: AnyHashable] {
+    var dictionary: [String: AnyHashable] = [:]
+    if let bucket = bucket {
+      dictionary["bucket"] = bucket
+    }
+    if let cacheControl = cacheControl {
+      dictionary["cacheControl"] = cacheControl
+    }
+    if let contentDisposition = contentDisposition {
+      dictionary["contentDisposition"] = contentDisposition
+    }
+    if let contentEncoding = contentEncoding {
+      dictionary["contentEncoding"] = contentEncoding
+    }
+    if let contentLanguage = contentLanguage {
+      dictionary["contentLanguage"] = contentLanguage
+    }
+    if let contentType = contentType {
+      dictionary["contentType"] = contentType
+    }
+    if let md5Hash = md5Hash {
+      dictionary["md5Hash"] = md5Hash
+    }
+    if let customMetadata = customMetadata {
+      dictionary["metadata"] = customMetadata
+    }
+    if size != 0 {
+      dictionary["size"] = size
+    }
+    if generation != 0 {
+      dictionary["generation"] = "\(generation)"
+    }
+    if metageneration != 0 {
+      dictionary["metageneration"] = "\(metageneration)"
+    }
+    if let timeCreated = timeCreated {
+      dictionary["timeCreated"] = StorageMetadata.RFC3339StringFromDate(timeCreated)
+    }
+    if let updated = updated {
+      dictionary["updated"] = StorageMetadata.RFC3339StringFromDate(updated)
+    }
+    if let path = path {
+      dictionary["name"] = path
+    }
+    return dictionary
```

Comment:
Good idea, but the compactMap doesn't work because of different types. Tests fail like
```
testUpdatedMetadataWithDelete(): XCTAssertEqual failed: ("["updated": AnyHashable(nil), "generation": AnyHashable("0"), "bucket": AnyHashable(nil), "timeCreated": AnyHashable(nil), "size": AnyHashable(0), "contentType": AnyHashable(nil), "md5Hash": AnyHashable(nil), "contentDisposition": AnyHashable(nil), "contentEncoding": AnyHashable(nil), "metageneration": AnyHashable("0"), "metadata": AnyHashable(["bar": AnyHashable(<null>)]), "name": AnyHashable(nil), "cacheControl": AnyHashable(nil), "contentLanguage": AnyHashable(nil)]") is not equal to ("["contentLanguage": AnyHashable(<null>), "metadata": AnyHashable(["bar": <null>])]")
```